src/services/bundleAnalysis/useBundleSummary.tsx

```diff
@@ -0,0 +1,190 @@
+  MissingHeadReportSchema,
+])
+
+const RepositorySchema = z.object({
+  __typename: z.literal('Repository'),
+  branch: z
+    .object({
+      head: z
+        .object({
+          bundleAnalysisReport: BundleReportSchema.nullable(),
+        })
+        .nullable(),
+    })
+    .nullable(),
+})
+
+const RequestSchema = z.object({
+  owner: z
+    .object({
+      repository: z
+        .discriminatedUnion('__typename', [
+          RepositorySchema,
+          RepoNotFoundErrorSchema,
+          RepoOwnerNotActivatedErrorSchema,
+        ])
+        .nullable(),
+    })
+    .nullable(),
+})
+
+const query = `
+query BundleSummary(
+  $owner: String!
+  $repo: String!
+  $branch: String!
+  $bundle: String!
+) {
+  owner(username: $owner) {
+    repository(name: $repo) {
+      __typename
+      ... on Repository {
+        branch(name: $branch) {
+          head {
+            bundleAnalysisReport {
+              __typename
+              ... on BundleAnalysisReport {
+                bundle(name: $bundle) {
+                  name
+                  moduleCount
+                  sizeTotal
+                  bundleData {
+                    loadTime {
+                      threeG
+                      highSpeed
+                    }
+                    size {
+                      gzip
+                      uncompress
+                    }
+                  }
+                }
+              }
+              ... on MissingHeadReport {
+                message
+              }
+            }
+          }
+        }
+      }
+      ... on NotFoundError {
+        message
+      }
+      ... on OwnerNotActivatedError {
+        message
+      }
+    }
+  }
+}`
+
+interface UseBundleSummaryArgs {
+  provider: string
+  owner: string
+  repo: string
+  branch: string
+  bundle: string
+}
+
+export const useBundleSummary = ({
+  provider,
+  owner,
+  repo,
+  branch,
+  bundle,
+}: UseBundleSummaryArgs) => {
+  return useQuery({
+    queryKey: ['BundleSummary', provider, owner, repo, branch, bundle],
+    queryFn: ({ signal }) =>
+      Api.graphql({
+        provider,
+        query,
+        signal,
+        variables: { owner, repo, branch, bundle },
+      }).then((res) => {
+        const parsedData = RequestSchema.safeParse(res?.data)
+
+        if (!parsedData.success) {
+          return Promise.reject({
+            status: 404,
+            data: {},
+            dev: 'useBundleSummary - 404 Failed to parse data',
+          } satisfies NetworkErrorObject)
+        }
+
+        const data = parsedData.data
+
+        if (data?.owner?.repository?.__typename === 'NotFoundError') {
+          return Promise.reject({
+            status: 404,
+            data: {},
+            dev: 'useBundleSummary - 404 Not found error',
+          } satisfies NetworkErrorObject)
+        }
+
+        if (data?.owner?.repository?.__typename === 'OwnerNotActivatedError') {
+          return Promise.reject({
+            status: 403,
```

Comment:
nit: this one might be better as a 401 instead of 403 as well

Reply:
For sure, the detail to get the user to Activate made me think it was more in line with unauthorized than forbidden though since the user _can_ fix it. I can see either way